File: packages/shared-ts/src/flight/state.ts
import { z } from 'zod';
import { FlightPositionSchema } from './position';
import { FlightVelocitySchema } from './velocity';
import { FlightPhaseSchema } from './phase';

export const FlightStateSchema = z.object({
  id: z.string(),
  callSign: z.string(),
  airline: z.string(),
  departureAirport: z.string(),
  arrivalAirport: z.string(),
  phase: FlightPhaseSchema,
  position: FlightPositionSchema,
  velocity: FlightVelocitySchema,
  bearing: z.number(),
  speed: z.number(),
  altitude: z.number(), // Duplicate of position.altitude for compatibility
  progress: z.number().min(0).max(1),
  distanceRemaining: z.number().min(0),
  scheduledDeparture: z.string().datetime(),
  scheduledArrival: z.string().datetime(),
  estimatedArrival: z.string().datetime(),
  lastComputedAt: z.string().datetime(),
  traceID: z.string(),
});

export type FlightState = z.infer<typeof FlightStateSchema>;

File: packages/shared-go/flight/state.go
package flight

type State struct {
	ID                 string   `json:"id"`
	CallSign           string   `json:"callSign"`
	Airline            string   `json:"airline"`
	DepartureAirport   string   `json:"departureAirport"`
	ArrivalAirport     string   `json:"arrivalAirport"`
	Phase              Phase    `json:"phase"`
	Position           Position `json:"position"`
	Velocity           Velocity `json:"velocity"`
	Bearing            float64  `json:"bearing"`
	Speed              float64  `json:"speed"`
	Altitude           float64  `json:"altitude"`
	Progress           float64  `json:"progress"`
	DistanceRemaining  float64  `json:"distanceRemaining"`
	ScheduledDeparture string   `json:"scheduledDeparture"`
	ScheduledArrival   string   `json:"scheduledArrival"`
	EstimatedArrival   string   `json:"estimatedArrival"`
	LastComputedAt     string   `json:"lastComputedAt"`
	TraceID            string   `json:"traceID"`
}
